src/modules/movie/movie.service.ts

```diff
@@ -20,7 +24,121 @@ export default class MovieService implements MovieServiceInterface {
     return result;
   }
 
-  public async findById(movieId: string): Promise<DocumentType<MovieEntity> | null> {
-    return this.movieModel.findById(movieId).exec();
+  public async findById(movieId: string): Promise<DocumentType<MovieEntity>[] | null> {
+    return this.movieModel
+      .aggregate([
+        {
+          $match: {'_id': new mongoose.Types.ObjectId(movieId)},
+        },
+        {
+          $lookup: {
+            from: 'comments',
+            localField: '_id',
+            foreignField: 'movieId',
+            as: 'commentData'
+          }
+        },
+        {
+          $set:
+          { rating: {$avg: '$commentData.rating'}, comments: { $size: '$commentData'}}
+        },
+        {
+          $lookup: {
+            from: 'users',
+            localField: 'userId',
+            foreignField: '_id',
+            as: 'user'
+          }
+        },
+        {
+          $unwind: '$user'
+        },
+        {
+          $unset: 'user._id'
+        },
+      ]);
   }
+
+  public async find(count?: number): Promise<DocumentType<MovieEntity>[]> {
+    const limit = count ?? DEFAULT_MOVIE_COUNT;
+    return this.movieModel
+      .find({}, {}, {limit})
+      .select(FILM_SHORT_FIELDS)
+      .sort({postDate: SortType.Down})
+      .populate(['userId'])
+      .exec();
+  }
+
+  public async editById(movieId: string, dto: EditMovieDto): Promise<DocumentType<MovieEntity> | null> {
+    return this.movieModel
+      .findByIdAndUpdate(movieId, dto, {new: true})
+      .populate(['userId'])
+      .exec();
+  }
+
+  public async deleteById(movieId: string): Promise<DocumentType<MovieEntity> | null> {
+    return this.movieModel
+      .findByIdAndDelete(movieId)
+      .exec();
+  }
+
+  public async findByGenre(genreType: string, count?: number): Promise<DocumentType<MovieEntity>[]> {
+    const limit = count ?? DEFAULT_MOVIE_COUNT;
+    return this.movieModel
+      .find({genre: genreType}, {}, {limit})
+      .sort({postDate: SortType.Down})
+      .populate(['userId'])
+      .exec();
+  }
+
+  public async findPromo(): Promise<DocumentType<MovieEntity>[] | null> {
+
+    return this.movieModel
+      .aggregate([
+        {
+          $sample: {size: 1},
+        },
+        {
+          $lookup: {
+            from: 'users',
+            localField: 'userId',
+            foreignField: '_id',
+            as: 'user'
+          }
+        },
+        {
+          $unwind: '$user'
+        },
+        {
+          $unset: 'user._id'
+        },
+      ]);
+  }
+
+  public async findFavorites(): Promise<DocumentType<MovieEntity>[] | null> { //WIP
+    return this.movieModel
+      .find()
+      .populate(['userId'])
+      .exec();
+  }
+
+  public async editFavorite(movieId: string): Promise<DocumentType<MovieEntity> | null> { //WIP
+    return this.movieModel
+      .findById(movieId)
+      .populate(['userId'])
```

Comment:
либо можно с другой стороны зайти и делать как то так
```suggestion
    const user = await this.userService.findById(userId);

    if (!user || !user.favorites.length) {
      return [];
    }
    return this.movieModel.aggregate([
      {$match: { _id: {
        $in: user.favorites.map((id) => new mongoose.Types.ObjectId(id))
      }}},
    ]);
```